docs/introduction-to-smart-contracts.rst

```diff
@@ -352,15 +356,19 @@ If the gas is used up at any point (i.e. it is negative),
 an out-of-gas exception is triggered, which reverts all modifications
 made to the state in the current call frame.
 
+Any unused gas is refunded at the end of the transaction.
+
 .. index:: ! storage, ! memory, ! stack
 
 Storage, Memory and the Stack
 =============================
 
-Each account has a persistent memory area which is called **storage**.
+The Ethereum Virtual Machine has three areas where it can store items.
+
+Each account has a memory area called **storage**, which is persistent between function calls.
 Storage is a key-value store that maps 256-bit words to 256-bit words.
 It is not possible to enumerate storage from within a contract
-and it is comparatively costly to read and even more so, to modify
+and it is comparatively costly to read and to modify
```

Comment:
Modifying is much more expensive than reading.

Reply:
@chriseth done